lib/spawn/cluster/state_handoff_supervisor.ex

```diff
@@ -7,10 +7,17 @@ defmodule Spawn.Cluster.StateHandoff.Supervisor do
     Supervisor.start_link(__MODULE__, state, name: __MODULE__)
   end
 
+  def child_spec(config) do
+    %{
+      id: __MODULE__,
+      start: {__MODULE__, :start_link, [config]}
+    }
+  end
+
   @impl true
-  def init(_args) do
-    children = [Spawn.Cluster.StateHandoff]
+  def init(config) do
+    children = [Spawn.Cluster.StateHandoff.child_spec(config)]
 
-    Supervisor.init(children, strategy: :one_for_one)
+    Supervisor.init(children, strategy: :one_for_one, max_restarts: 100)
```

Comment:
As you increased the number of restarts I would also increase the timeframe using a larger value for max_seconds. As this is a vital process I would also increase it to a value greater than 100. Both max_restarts and max_seconds should be configurable via config.